src/Processors/Formats/Impl/ArrowColumnToCHColumn.cpp

```diff
@@ -564,21 +566,39 @@ Block ArrowColumnToCHColumn::arrowSchemaToCHHeader(
     return Block(std::move(sample_columns));
 }
 
-ArrowColumnToCHColumn::ArrowColumnToCHColumn(
-    const Block & header_,
-    const std::string & format_name_,
-    bool import_nested_,
-    bool allow_missing_columns_,
-    bool case_insensitive_matching_)
+ArrowColumnToCHColumn::ArrowColumnToCHColumn(const Block & header_, std::shared_ptr<arrow::Schema> schema_, const std::string & format_name_, const FormatSettings & settings_)
     : header(header_)
+    , schema(std::move(schema_))
     , format_name(format_name_)
-    , import_nested(import_nested_)
-    , allow_missing_columns(allow_missing_columns_)
-    , case_insensitive_matching(case_insensitive_matching_)
+    , null_as_default(settings_.null_as_default)
+    , defaults_for_omitted_fields(settings_.defaults_for_omitted_fields)
 {
+    if (format_name == "ORC")
+    {
+        import_nested = settings_.orc.import_nested;
+        allow_missing_columns = settings_.orc.allow_missing_columns;
+        case_insensitive_matching = settings_.orc.case_insensitive_column_matching;
+    }
+    else if (format_name == "Parquet")
+    {
+        import_nested = settings_.parquet.import_nested;
+        allow_missing_columns = settings_.parquet.allow_missing_columns;
+        case_insensitive_matching = settings_.parquet.case_insensitive_column_matching;
+    }
+    else if (format_name == "Arrow")
+    {
+        import_nested = settings_.arrow.import_nested;
+        allow_missing_columns = settings_.arrow.allow_missing_columns;
+        case_insensitive_matching = settings_.arrow.case_insensitive_column_matching;
+    }
+    else
+    {
+        throw Exception(ErrorCodes::UNKNOWN_FORMAT, "Unsupported format {}", format_name);
+    }
```

Comment:
What was the purpose to do this instead of providing these settings as constructor arguments? I would prefer previous implementation